pkg/service/keystore/storage.go

```diff
@@ -48,43 +56,107 @@ const (
 )
 
 type Storage struct {
-	db         storage.ServiceStorage
-	serviceKey []byte
+	db        storage.ServiceStorage
+	encryptor Encryptor
+	decryptor Decryptor
 }
 
-func NewKeyStoreStorage(db storage.ServiceStorage, key ServiceKey) (*Storage, error) {
-	keyBytes, err := base58.Decode(key.Base58Key)
+func NewKeyStoreStorage(db storage.ServiceStorage, e Encryptor, d Decryptor) (*Storage, error) {
+	bolt := &Storage{
+		db:        db,
+		encryptor: e,
+		decryptor: d,
+	}
+
+	return bolt, nil
+}
+
+type wrappedEncryptor struct {
+	tink.AEAD
+}
+
+func (w wrappedEncryptor) Encrypt(_ context.Context, plaintext, contextData []byte) ([]byte, error) {
```

Comment:
why take in a context?

Reply:
Because `wrappedEncryptor` needs to implement the `Encryptor` interface